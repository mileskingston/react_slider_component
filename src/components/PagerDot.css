.slider__pager {
  display: inline-block;
}
.slider__pager .pager__dot:first-child {
  margin-left: 0;
}
.slider__pager .pager__dot:last-child {
  margin-right: 0;
}

.pager__dot {
  display: inline-block;
  width: 10px;
  height: 10px;
  padding: 0;
  margin: 0 3px;
  border: 0;
  border-radius: 50%;
  background: grey;
}
.pager__dot--active {
  background: black;
}