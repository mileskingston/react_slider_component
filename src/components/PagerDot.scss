@import "../_variables.scss";

.slider__pager {
  display: inline-block;
  
  .pager__dot:first-child {
    margin-left: 0;
  }
  .pager__dot:last-child {
    margin-right: 0;
  }
}

.pager__dot {
  display: inline-block;
  width: 10px;
  height: 10px;
  padding: 0;
  margin: 0 3px;
  border: 0;
  border-radius: 50%;
  background: $grey;
  cursor: pointer;
  
  &:hover {
    background: $grey-dark;
  }

  &--active {
    background: $black;
  }
}