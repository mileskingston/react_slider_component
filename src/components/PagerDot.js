import React from 'react';
import PropTypes from 'prop-types';

import './PagerDot.scss';

const PagerDot = ({
  onClick,
  currentSlide,
  slide
}) => (
  <button
    type="button"
    className={`pager__dot ${currentSlide ? 'pager__dot--active' : ''}`}
    onClick={() => { onClick(); }}
    tabIndex={currentSlide ? '-1' : '0'}
  >
    <span className="sr--only">{slide}</span>
  </button>
);

PagerDot.displayName = 'PagerDot';

PagerDot.propTypes = {
  currentSlide: PropTypes.bool.isRequired,
  onClick: PropTypes.func.isRequired,
  slide: PropTypes.number.isRequired
};

export default PagerDot;
