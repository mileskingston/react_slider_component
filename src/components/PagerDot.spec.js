import React from 'react';
import { shallow } from 'enzyme';

import PagerDot from './PagerDot';

const curruentSlideTrue = true;

describe('PagerDot component', () => {
    it('should render with matching props', () => {
        const mockFunc = jest.fn();
        const wrapper = shallow(
            <PagerDot slide={0} currentSlide={curruentSlideTrue} onClick={mockFunc} />
        );
        
        expect(wrapper).toMatchSnapshot();
    });

    it('should call onClick function on click', () => {
        const mockFunc = jest.fn();
        const wrapper = shallow(<PagerDot slide={0} currentSlide={curruentSlideTrue} onClick={mockFunc} />);

        wrapper.simulate('click');
        expect(mockFunc).toHaveBeenCalled();
    });
}); 