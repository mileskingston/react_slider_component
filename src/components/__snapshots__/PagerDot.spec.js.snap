// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PagerDot component should render with matching props 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <PagerDot
    currentSlide={true}
    onClick={[MockFunction]}
    slide={0}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <span
        className="sr--only"
      >
        0
      </span>,
      "className": "pager__dot pager__dot--active",
      "onClick": [Function],
      "tabIndex": "-1",
      "type": "button",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": 0,
        "className": "sr--only",
      },
      "ref": null,
      "rendered": 0,
      "type": "span",
    },
    "type": "button",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <span
          className="sr--only"
        >
          0
        </span>,
        "className": "pager__dot pager__dot--active",
        "onClick": [Function],
        "tabIndex": "-1",
        "type": "button",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": 0,
          "className": "sr--only",
        },
        "ref": null,
        "rendered": 0,
        "type": "span",
      },
      "type": "button",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
